### PR TITLE
test for retrieving exit code

### PR DESCRIPTION
**Pull request by zhy76**
test for retrieving exit code
Fixes: #634

### CODE REVIEW

**Comment by zhy76** [comment]
cc @kkourt PTAL, Thanks!

**Review by tixxdz — COMMENTED** [review]
Thank you! some comments below ;-)

File: pkg/sensors/exec/exit_test.go [hunk 1]
```diff
@@ -173,3 +173,70 @@ func TestExitZombie(t *testing.T) {
...
+		t.Fatalf("Failed to run observer: %s", err)
+	}
+	observer.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
+	readyWG.Wait()
+
+	testExitCode := testutils.RepoRootPath("contrib/tester-progs/exit-code")
+
+	var exitCode, realCode uint32
+
+	if err := exec.Command(testExitCode).Run(); err != nil {
```
[the remark this answers is not in the document]
Comment (zhy76) [reply pkg/sensors/exec/exit_test.go:203]: Thanks for you suggestions! I'll update my code.

File: pkg/sensors/exec/exit_test.go [hunk 2]
```diff
@@ -173,3 +173,70 @@ func TestExitZombie(t *testing.T) {
...
+			t.Fatalf("Failed to execute test binary: %s\n", err)
+		}
+	}
+
+	// The test executes 'exit-code' benary which return a exit code 6
+	nextCheck := func(event ec.Event, l *logrus.Logger) (bool, error) {
+		switch ev := event.(type) {
+		case *tetragon.ProcessExit:
+			if ev.Process.Binary == testExitCode {
+				exitCode = ev.Status
```
[the remark this answers is not in the document]
Comment (zhy76) [reply pkg/sensors/exec/exit_test.go]: I run the process before tetragon, the experientment shows that we can't get the "status" field.

**Comment by zhy76** [comment]
> Then later in an other PR in your exit-code you may also try to make it signal to itself and ensure that the field "signal": "SIGINT" example in the process_exit event is set as an example, then you can combine with the exit code too, etc.
I'll add this in an other pr. PTAL, thanks! @tixxdz

**Comment by jrfastab** [comment]
I'm happy with this so will merge it. Please address any follup from reviewers if they come back with feedback.

**Pull request merged**